pkg/ovs/openflow/ofctrl_builder_test.go

```diff
@@ -15,30 +15,2171 @@
+				ip:                 net.ParseIP("1.1.1.1"),
+				expectedMatcherStr: "ct_nw_src=1.1.1.1",
+				expectedMatchField: &matchField{
+					class: openflow15.OXM_CLASS_NXM_1,
+					field: openflow15.NXM_NX_CT_NW_SRC,
+					value: &openflow15.Ipv4SrcField{Ipv4Src: net.ParseIP("1.1.1.1")},
+				},
+			},
+			{
+				isSrc:              true,
+				ip:                 net.ParseIP("fec0::1111"),
+				expectedMatcherStr: "ct_ipv6_src=fec0::1111",
+				expectedMatchField: &matchField{
+					class: openflow15.OXM_CLASS_NXM_1,
+					field: openflow15.NXM_NX_CT_IPV6_SRC,
+					value: &openflow15.Ipv6SrcField{Ipv6Src: net.ParseIP("fec0::1111")},
+				},
+			},
+			{
+				ip:                 net.ParseIP("1.1.1.1"),
+				expectedMatcherStr: "ct_nw_dst=1.1.1.1",
+				expectedMatchField: &matchField{
+					class: openflow15.OXM_CLASS_NXM_1,
+					field: openflow15.NXM_NX_CT_NW_DST,
+					value: &openflow15.Ipv4DstField{Ipv4Dst: net.ParseIP("1.1.1.1")},
+				},
+			},
+			{
+				ip:                 net.ParseIP("fec0::1111"),
+				expectedMatcherStr: "ct_ipv6_dst=fec0::1111",
+				expectedMatchField: &matchField{
+					class: openflow15.OXM_CLASS_NXM_1,
+					field: openflow15.NXM_NX_CT_IPV6_DST,
+					value: &openflow15.Ipv6DstField{Ipv6Dst: net.ParseIP("fec0::1111")},
+				},
+			},
+		}
+		for _, tc := range testCases {
+			var fb *ofFlowBuilder
+			if tc.isSrc {
+				fb = table.BuildFlow(1).MatchCTSrcIP(tc.ip).(*ofFlowBuilder)
+				if tc.ip.To4() != nil {
+					assert.Equal(t, tc.ip, *fb.ofFlow.Match.CtIpSa)
+				} else {
+					assert.Equal(t, tc.ip, *fb.ofFlow.Match.CtIpv6Sa)
+				}
+			} else {
+				fb = table.BuildFlow(1).MatchCTDstIP(tc.ip).(*ofFlowBuilder)
+				if tc.ip.To4() != nil {
+					assert.Equal(t, tc.ip, *fb.ofFlow.Match.CtIpDa)
+				} else {
+					assert.Equal(t, tc.ip, *fb.ofFlow.Match.CtIpv6Da)
+				}
+			}
+
+			flowMod := getFlowMod(t, fb.Done())
+			assert.Equal(t, 1, len(flowMod.Match.Fields))
+			checkMatchField(t, tc.expectedMatchField, flowMod.Match.Fields[0])
+			assert.Contains(t, getFlowModMatch(flowMod), tc.expectedMatcherStr)
+		}
+	})
+	t.Run("MatchCTSrcIPNet/MatchCTDstIPNet", func(t *testing.T) {
+		_, ipv4Net1, _ := net.ParseCIDR("1.1.1.0/24")
+		_, ipv4Net2, _ := net.ParseCIDR("1.1.1.1/32")
+		_, ipv6Net1, _ := net.ParseCIDR("fe80::1111/64")
+		_, ipv6Net2, _ := net.ParseCIDR("fec0::ffff/128")
+		testCases := []struct {
+			isSrc              bool
+			ipNet              *net.IPNet
+			expectedMask       net.IP
+			expectedMatcherStr string
+			expectedMatchField *matchField
+		}{
+			{
+				isSrc:              true,
+				ipNet:              ipv4Net1,
+				expectedMask:       net.IP(ipv4Net1.Mask),
+				expectedMatcherStr: "ct_nw_src=1.1.1.0/24",
+				expectedMatchField: &matchField{
+					class:   openflow15.OXM_CLASS_NXM_1,
+					field:   openflow15.NXM_NX_CT_NW_SRC,
+					hasMask: true,
+					value:   &openflow15.Ipv4SrcField{Ipv4Src: ipv4Net1.IP},
+					mask:    &openflow15.Ipv4SrcField{Ipv4Src: net.IP(ipv4Net1.Mask)},
```

Comment:
You may define a few new variable for these masks which are called multiple times.
`net.IP(ipv4Net1.Mask)`, ` net.IP(ipv4Net2.Mask)` etc.

Reply:
Thanks, will do.